src/search-scenario/azext_search_scenario/custom.py

```diff
@@ -0,0 +1,312 @@
+        print()
+
+
+def _execute_scenario(ctx_cmd, scenario):
+
+    for command in scenario["commandSet"]:
+        parameters = []
+        if "arguments" in command:
+            parameters = command["arguments"]
+
+        if ctx_cmd.cli_ctx.config.getboolean('search_scenario', 'print_help', fallback=False):
+            _print_help_info(ctx_cmd, command["command"])
+
+        print_styled_text([(Style.ACTION, "Running: ")], end='')
+        print_styled_text(_get_command_item_sample(command))
+        option_msg = [(Style.ACTION, " ? "),
+                      (Style.PRIMARY, "How do you want to run this step? 1. Run it 2. Skip it 3. Quit process "),
+                      (Style.SECONDARY, "(Enter is to Run)"), (Style.PRIMARY, ": ")]
+        run_option = get_int_option(option_msg, 1, 3, 1)
+        if run_option == 1:
+            print_styled_text([(Style.SECONDARY, "Input Enter to skip unnecessary parameters")])
+            execute_result = _execute_cmd(ctx_cmd, command['command'], parameters, catch_exception=True)
+            is_help_printed = False
+            while execute_result != 0:
+                if not ctx_cmd.cli_ctx.config.getboolean('search_scenario', 'print_help', fallback=False) \
+                        and not is_help_printed:
+                    _print_help_info(ctx_cmd, command["command"])
+                    is_help_printed = True
+
+                option_msg = [(Style.ACTION, " ? "),
+                              (Style.PRIMARY, "Do you want to retry this step? 1. Run it 2. Skip it 3. Quit process "),
+                              (Style.SECONDARY, "(Enter is to Run)"), (Style.PRIMARY, ": ")]
+                run_option = get_int_option(option_msg, 1, 3, 1)
+                if run_option == 1:
+                    execute_result = _execute_cmd(ctx_cmd, command['command'], parameters, catch_exception=True)
+                elif run_option == 2:
+                    print()
+                    break
+                else:
+                    print()
+                    return
+
+        elif run_option == 2:
+            print()
+            continue
+        else:
+            print()
+            break
+
+    from .utils import print_successful_styled_text
+    print_successful_styled_text('All commands in this scenario have been executed! \n')
+
+
+def _style_highlight(highlight_content: str) -> list:
+    styled_description = []
+    remain = highlight_content
+    in_highlight = False
+    while remain:
+        split_result = remain.split('<em>' if not in_highlight else "</em>", 1)
+        content = split_result[0]
+        try:
+            remain = split_result[1]
+        except IndexError:
+            remain = None
+        styled_description.append((Style.SECONDARY if not in_highlight else Style.WARNING, content))
+        in_highlight = not in_highlight
+    return styled_description
+
+
+def _execute_cmd(ctx_cmd, command, params, catch_exception=False):
```

Comment:
Split function to small functions with meaningful function name or add comments to help reviewers understand.

Reply:
Done